modules/distributor/forwarder.go

```diff
@@ -77,25 +78,25 @@ func newForwarder(fn forwardFunc, o *overrides.Overrides) *forwarder {
 }
 
 // SendTraces queues up traces to be sent to the metrics-generators
-func (f *forwarder) SendTraces(ctx context.Context, tenantID string, keys []uint32, traces []*rebatchedTrace) {
+func (f *generatorForwarder) SendTraces(ctx context.Context, tenantID string, keys []uint32, traces []*rebatchedTrace) {
 	select {
 	case <-f.shutdown:
 		return
 	default:
 	}
 
-	qm := f.getOrCreateQueueManager(tenantID)
-	err := qm.pushToQueue(ctx, &request{keys: keys, traces: traces})
+	q := f.getOrCreateQueue(tenantID)
```

Comment:
this would be far easier to review if we had split the refactor from the addition of the new forwarder. is that still possible?

Reply:
Done.

1. Refactor: #1796
2. I've rebased this PR on top of #1796. It will become easier to review after we merge #1796 to origin/main.